### DISCUSSION TITLE
User & Password

### DISCUSSION THREAD

**Discussion by sdapestey** [jvde-github/AIS-catcher#261] [General]
Hello, is there a plugin or is there any possibility of entering a username and password when a user enters the site?

Thank you

**Comment by Phil353556** [comment]
Hi,
You don't want that https://aiscatcher.org stay accessible without any user/password ?
Why?   
May you elaborate?

**Reply by sdapestey** [reply]
I want it for my own station installed on a raspberry. Not for the aiscatcher.org site

**Reply by Phil353556** [reply]
ah ok.
Ok I have the same installation ie a raspberry pi.
This is on your local network right?
and you don't want to the website to be accessible without login/password even on your private network. right?

**Reply by sdapestey** [reply]
Yes sir.
It's for my local network. I want only certain users to be able to access the web server

**Reply by Phil353556** [reply]
ok, let us see what @jvde-github the Master of aiscatcher will say about that.

**Reply by jvde-github** [reply]
There is no option as such. The best approach is to put it behind a nginx reverse proxy and set up a password there. Also, as it is not https it is not secure even with a password. 

Having said that, have been thinking about the topic a while ago as it is useful when we let the user configure the program (e.g. UDP output) via the web interface. So might add it one day.... Unfortunately, for now, I think the nginx route?

**Reply by sdapestey** [reply]
I'm going to use "auth_basic_user_file /etc/nginx/.htpasswd;"

Thanks for support.

**Comment by jvde-github** [comment]
I also added a new option to -N that would allow you to bind the webviewer to a specific IP, e.g. `-N 8100 IP_BIND localhost`

This would prevent people accessing the webviewer from another PC. Otherwise you have to block it via a firewall. WIll have to still ad in the docs and check it for windows.
